File: manager/lib/features/voucher/repository/dto/voucher_request.dart
class VoucherRequest {
  final String name;
  final double discountAmount;
  final DateTime startDate;
  final DateTime endDate;
  final List<int> productIds;

  VoucherRequest({
    required this.name,
    required this.discountAmount,
    required this.startDate,
    required this.endDate,
    required this.productIds,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'discountAmount': discountAmount,
      'startDate': startDate.toIso8601String(),
      'endDate': endDate.toIso8601String(),
      'productIds': productIds,
    };
  }
}

File: Backend/src/main/java/com/example/backend/dto/VoucherRequest.java
package com.example.backend.dto;

import java.time.LocalDateTime;
import java.util.List;
import lombok.Data;

@Data
public class VoucherRequest {
    private String name;
    private Double discountAmount;
    private LocalDateTime startDate;
    private LocalDateTime endDate;

    private List<Long> productIds;
}